src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -2569,6 +2571,67 @@ create_exp_macro(Exp, exp)
 create_exp_macro(Exp2, exp2)
 create_exp_macro(Expm1, expm1)
 
+namespace Input {
+
+    static inline void verify_args(const ASR::IntrinsicScalarFunction_t& x,
+            diag::Diagnostics& diagnostics) {
+        ASRUtils::require_impl(x.n_args <= 1,
+            "ASR Verify: Call `input` takes 0 or 1 argument",
+            x.base.base.loc, diagnostics);
+    }
+
+    static inline ASR::asr_t* create_Input(Allocator& al, const Location& loc,
```

Comment:
Let's take this function and remove it from the IntrinsicScalarFunction registory, and put it into some utility file in libasr.

Then call it from the frontend.

Reply:
@certik What can be a good place to add it? 

Also, should we instead implement this through an ASR node? We can implement it exactly in the way `print()` is handled. Please guide me.

Reply:
For now put this `create_Input` into `src/libasr/asr_utils.cpp`. We can later move it elsewhere if we find a better place.

I would implement it exactly as you did just now, but make it a function in `asr_utils.cpp` and call it from the frontend. So ASR does not need to be modified. We can later improve all the IO nodes in ASR, but I want to use the same nodes for all the frontends.